From foo@baz Tue Apr  9 12:12:43 2002
Date: Fri, 23 May 2008 09:44:11 -0700
To: Greg KH <greg@kroah.com>
From: Randy Dunlap <randy.dunlap@oracle.com>
Subject: block: make /proc/diskstats only build if CONFIG_PROC_FS is enabled

These functions are only needed if CONFIG_PROC_FS is enabled, so save
the space when it is not.

This also makes it easier for a patch later in this series to work
properly if CONFIG_PROC_FS is not enabled.

Signed-off-by: Randy Dunlap <randy.dunlap@oracle.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 block/genhd.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/block/genhd.c
+++ b/block/genhd.c
@@ -532,6 +532,7 @@ static struct device_type disk_type = {
 	.release	= disk_release,
 };
 
+#ifdef CONFIG_PROC_FS
 /*
  * aggregate disk stat collector.  Uses the same stats that the sysfs
  * entries do, above, but makes them available through one seq_file.
@@ -641,6 +642,7 @@ const struct seq_operations diskstats_op
 	.stop	= diskstats_stop,
 	.show	= diskstats_show
 };
+#endif /* CONFIG_PROC_FS */
 
 static void media_change_notify_thread(struct work_struct *work)
 {
